cd/auto-generate/auto-generate-controllers.sh

```diff
@@ -0,0 +1,241 @@
+                       be created when autogeneration of service controller fails.
+                       Defaults to 'aws-controllers-k8s'
+  GH_ISSUE_REPO:       Name of the GitHub repository where GitHub issues will
+                       be created when autogeneration of service controller fails.
+                       Defaults to 'community'
+  GH_LABEL:            Label to add to issue and pull requests.
+                       Defaults to 'ack-bot-autogen'
+  GITHUB_ACTOR:        Name of the GitHub account creating the issues & PR.
+  GITHUB_DOMAIN:       Domain for GitHub. Defaults to 'github.com'
+  GITHUB_EMAIL_PREFIX: The 7 digit unique id for no-reply email of
+                       '$GITHUB_ACTOR'
+  GITHUB_TOKEN:        Personal Access Token for '$GITHUB_ACTOR'
+"
+
+# Important Directory references based on prowjob configuration.
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+AUTO_GEN_DIR=$THIS_DIR
+CD_DIR=$THIS_DIR/..
+TEST_INFRA_DIR=$CD_DIR/..
+WORKSPACE_DIR=$TEST_INFRA_DIR/..
+CODEGEN_DIR=$WORKSPACE_DIR/code-generator
+
+DEFAULT_PR_SOURCE_BRANCH="ack-bot-autogen"
+PR_SOURCE_BRANCH=${PR_SOURCE_BRANCH:-$DEFAULT_PR_SOURCE_BRANCH}
+
+DEFAULT_PR_TARGET_BRANCH="main"
+PR_TARGET_BRANCH=${PR_TARGET_BRANCH:-$DEFAULT_PR_TARGET_BRANCH}
+
+LOCAL_GIT_BRANCH="main"
+
+DEFAULT_GH_ISSUE_ORG="aws-controllers-k8s"
+GH_ORG=${GH_ORG:-$DEFAULT_GH_ISSUE_ORG}
+
+DEFAULT_GH_ISSUE_REPO="community"
+GH_ISSUE_REPO=${GH_ISSUE_REPO:-$DEFAULT_GH_ISSUE_REPO}
+
+DEFAULT_GH_LABEL="ack-bot-autogen"
+GH_LABEL=${GH_LABEL:-$DEFAULT_GH_LABEL}
+
+GITHUB_EMAIL_PREFIX="82905295"
+
+# Check all the dependencies are present in container.
+source "$TEST_INFRA_DIR"/scripts/lib/common.sh
+check_is_installed git
+check_is_installed gh
+
+USER_EMAIL="${GITHUB_ACTOR}@users.noreply.${GITHUB_DOMAIN:-"github.com"}"
+if [ -n "${GITHUB_EMAIL_PREFIX}" ]; then
+    USER_EMAIL="${GITHUB_EMAIL_PREFIX}+${USER_EMAIL}"
+fi
+
+# set the GitHub configuration for using GitHub cli.
+git config --global user.name "${GITHUB_ACTOR}" >/dev/null
+git config --global user.email "${USER_EMAIL}" >/dev/null
+
+# Findout the runtime semver from the code-generator repo
+cd "$CODEGEN_DIR"
+ACK_RUNTIME_VERSION=$(grep "github.com/aws-controllers-k8s/runtime" go.mod | grep -oE "v[0-9]+\.[0-9]+\.[0-9]+")
+if [[ -z $ACK_RUNTIME_VERSION ]]; then
+  echo "auto-generate-controllers.sh][ERROR] Unable to determine ACK runtime version from code-generator/go.mod file. Exiting."
+  exit 1
+else
+  echo "auto-generate-controllers.sh][INFO] ACK runtime version for new controllers will be $ACK_RUNTIME_VERSION"
+fi
+
+# find all the directories whose name ends with 'controller'
+pushd "$WORKSPACE_DIR" >/dev/null
+  CONTROLLER_NAMES=$(find . -maxdepth 1 -mindepth 1 -type d | cut -d"/" -f2 | grep -E "controller$")
+popd >/dev/null
+
+for CONTROLLER_NAME in $CONTROLLER_NAMES; do
+  SERVICE_NAME=$(echo "$CONTROLLER_NAME"| sed 's/-controller$//g')
```

Comment:
Could do `echo "$CONTROLLER_NAME" | cut -d"-" -f1` as well

Reply:
Did this specifically to avoid scenarios if a service name has '-' in it. IMO, Better to just remove the "-controller" suffix.

Reply:
It would be against a bunch of naming conventions for any service to have a hyphen in them. None of the current aws-sdk-go packages contain that - https://docs.aws.amazon.com/sdk-for-go/api/